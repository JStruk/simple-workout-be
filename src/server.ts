import { config } from 'dotenv'
config()

import Koa from 'koa'
import cors from '@koa/cors'

import router from './routes/router'

export const DEFAULT_PORT = 3001

const bodyParser = require('koa-bodyparser')

export function startServer() {
	const port = process.env.APP_PORT || DEFAULT_PORT

	const app = new Koa()
	app.use(cors())
	app.use(bodyParser())
	app.use(router.routes())

	if (process.env.NODE_ENV !== 'test') {
		return app.listen(port, () => {
			console.log(`Server running on port ${port}`)
		})
	}
	return
}

export default startServer()
