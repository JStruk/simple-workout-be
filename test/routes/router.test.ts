import request from 'supertest'
import server from '../../src/server'

describe('router', () => {
	afterAll(() => {
		server.close()
	})

	describe('home', () => {
		it('returns hello world', async () => {
			const response = await request(server).get('/test')
			expect(response.status).toBe(200)
			expect(response.text).toMatchSnapshot()
		})
	})
})