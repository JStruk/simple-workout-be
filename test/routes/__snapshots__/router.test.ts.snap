// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router home returns some shit 1`] = `"Hello World!"`;
